File: src/app/services/bills.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Bill } from '../bills';
import { Payment } from '../payment';

@Injectable({
  providedIn: 'root'
})
export class BillsService {

  private apiUrl = 'http://localhost:8090/api/bills';
  private apiUrl2 = 'http://localhost:8090/api/payments';

  constructor(private http: HttpClient) {}

  getBillsByCustomerId(customerId: number): Observable<any[]> {
    return this.http.get<any[]>(`${this.apiUrl}`);
  }
 addBill(bill:any,customerId:any)
 {
  return this.http.post(`${this.apiUrl}/create/${customerId}`,bill);
 }
 
 getBillById(billId: any): Observable<Bill> {
  return this.http.get<Bill>(`${this.apiUrl}/${billId}`);
}

// Update the bill and its payment status
updateBill(payment: Payment): Observable<Payment> {
  // Assuming the payment info is sent to update the bill's payment status
  return this.http.post<Payment>(`${this.apiUrl2}`, payment);
}
 
  
}

File: src/main/java/com/example/demo/service/BillsService.java
package com.example.demo.service;





import java.util.List;

import com.example.demo.model.Bills;

public interface BillsService {
    List<Bills> getAllBills(); // Fetch all bills
    List<Bills> getBillsByCustomerId(Long customerId); // Fetch bills by customer ID
    Bills saveBill(Bills bill,long customerId); // Create or update a bill
    Bills getBillById(Long billId); // Get a specific bill by ID
    Bills updateBill(Long billId, Bills updatedBill); // Update a bill
    void deleteBill(Long billId); // Delete a bill
}
